### PR TITLE
Editor: Replaced ffmpeg.js with WebCodecs API + mp4box.js

### PR DESCRIPTION
**Pull request by mrdoob**
**Description**

This PR replaces ffmpeg.js with the [WebCodecs API](https://caniuse.com/webcodecs) and [mp4box.js](https://github.com/gpac/mp4box.js/).
Videos are created *much* faster and we no longer need the [Service Worker hack](https://github.com/mrdoob/three.js/pull/24217).

The mp4 it produces [seem to upload to Twitter without problems](https://twitter.com/mrdoob/status/1615300387887796224), but seems like the output quality is much lower and the `bitrate` parameter in `VideoEncoder` doesn't seem to affect much... 🤔

If anyone wants to mess with the options, this tool is very useful for analyzing the files: 
https://gpac.github.io/mp4box.js/test/filereader.html

### CODE REVIEW

**Comment by LeviPesin** [comment]
Seems like WebCodecs are not supported in Firefox and Safari yet?

File: editor/js/Sidebar.Project.Video.js [hunk 1]
```diff
@@ -70,73 +70,108 @@ function SidebarProjectVideo( editor ) {
...
-		let currentTime = 0;
+				if ( trackID === null ) {
 
-		for ( let i = 0; i < frames; i ++ ) {
+					trackID = mp4.addTrack( {
+						timescale: 1_000 * 1_000,
+						width: width,
+						height: height,
+						nb_samples: frames,
+						media_duration: duration * 1_000 * 1_000,
```
Comment (LeviPesin) [inline editor/js/Sidebar.Project.Video.js:98]: I think numeric separators like in `1_000` aren't supported very much yet...

File: editor/js/Sidebar.Project.Video.js [hunk 2]
```diff
@@ -70,73 +70,108 @@ function SidebarProjectVideo( editor ) {
...
+		const duration = videoDuration.getValue();
+		const fps = videoFPS.getValue();
+		const frames = duration * fps;
+		const frameDuration = Math.floor( 1_000_000 / fps );
 
-		ffmpeg.setProgress( ( { ratio } ) => {
+		//
 
-			progress.setValue( ( ratio * 0.5 ) + 0.5 );
+		const mp4 = MP4Box.createFile();
```
Comment (LeviPesin) [inline editor/js/Sidebar.Project.Video.js:83]:
```suggestion
		const mp4 = MP4Box.createFile(); // eslint-disable-line no-undef
```

File: editor/js/Sidebar.Project.Video.js (hunk 1 above)
Comment (mrpeu) [reply editor/js/Sidebar.Project.Video.js:98]: Pretty good suuport actually: https://caniuse.com/mdn-javascript_grammar_numeric_separators

**Comment by mrdoob** [comment]
> Seems like WebCodecs are not supported in Firefox and Safari yet?

Seems to be supported in Safari TP (but I have not tested it yet). Unfortunately, rendering to video won't be supported in Firefox until they implement WebCodecs.

The thing is that the current Service Workers hack made things confusing... Sometimes the panel would be there after reloading the page, some times it wouldn't. Hard to understand why. Better to just have it working 100% of the times even if it's not in all browsers.

Also, the render speed is waaaay faster.

**Comment by mrdoob** [comment]
Just tried in Safari TP. Seems like they don't support `canvas` in `VideoFrame` yet.

Screenshot 2023-01-18 at 10 46 21

**Comment by mrdoob** [comment]
I've researched this a bit more...

Seems like `VideoEncoder` can't do more than `bitrate: 10_000_000`.
Any value bigger than that give no visual improvement.

Better to put this on hold until that problem is solved.

**Comment by zhaoran112** [comment]
Can mp4box.js export video files with transparent background?  @mrdoob

**Pull request closed without merging**